app/database_etl/location_utils/location_util_functions.py

```diff
@@ -76,4 +81,66 @@ def get_city(row):
             return [f"{city},{row['state'][0]}" for city in cities]
         return cities
     else:
-        return row['city']
\ No newline at end of file
+        return row['city']
+
+# Checks whether each row of an input city or state dataframe
+# is in one of the WHO's disputed areas
+def check_if_in_disputed_area(df: pd.DataFrame) -> pd.DataFrame:
+    WHO_FL_URL = "https://services.arcgis.com/5T5nSi527N4F7luB/arcgis/rest/services/DISPUTED_AREAS_mask/FeatureServer/0"
+    # Create feature layer object
+    disputed_areas_fl = FeatureLayer(WHO_FL_URL)
+
+    # Defining this as a closure so that we have access to
+    # disputed_areas_fl
+    def row_in_disputed_area(row: pd.Series) -> bool:
+        # Null check
+        if pd.isna(row['pin_longitude']) or pd.isna(row['pin_latitude']):
+            return False
+        # Construct a point at the row's coordinates
+        pin = Point({"x": row['pin_longitude'], "y": row['pin_latitude']})
+        # construct a geometry filter to check if each point is in a disputed area
+        pin_filter = intersects(pin)
+        in_disputed_area = len(disputed_areas_fl.query(geometry_filter=pin_filter).features) > 0
+        return in_disputed_area
+
+    # apply row_in_disputed_area across the whole df
+    df['in_disputed_area'] = df.apply(lambda row: row_in_disputed_area(row), axis=1)
+    return df
+
+
+# Get pin latitude and longitude for a record
+# Param geo_dfs = dictionary of city, state, and country DFs
+def get_record_coordinates(record: pd.Series, geo_dfs: dict) -> Tuple:
```

Comment:
can we just define geo_dfs in here? why are we passing it in all the way from etl_main

Reply:
geo_dfs contains the dfs for the city, state, and country tables. We don't have access to those tables here, hence we must pass them from etl_main